oauthenticator.py

```diff
@@ -216,10 +220,38 @@ def authenticate(self, handler):
         resp_json = json.loads(resp.body.decode('utf8', 'replace'))
 
         username = resp_json["username"]
-        if self.whitelist and username not in self.whitelist:
+        whitelisted = yield self.check_whitelist(username, headers)
+        if not whitelisted:
             username = None
         raise gen.Return(username)
 
+    @gen.coroutine
+    def check_whitelist(self, username, headers):
+        if self.team_whitelist:
+            valid_user = yield self._check_group_whitelist(username, headers)
+        else:
+            valid_user = yield super().check_whitelist(username)
+        raise gen.Return(valid_user)
```

Comment:
We can use `return valid_user` here and elsewhere, since we don't need to support legacy Python.